deployment/ansible/mnesia_backup.yml

```diff
@@ -0,0 +1,60 @@
+---
+# Playbook used to download (archived) Mnesia backup from last host in the playbook hosts
+#
+# By default the archive destination is overwritten if already exists,
+# if archive overwrite is not intended the caller is responsible for setting correct (e.g. unique) "backup_suffix"
+#
+# Example steps to restore a mnesia backup:
+# 1. Stop the node
+# 2. Delete existing mnesia database if any
+# 3. Disable mining autostart and empty the peers list in the config
+# 4. Start the node
+# 5. Restore the backup with "mnesia:restore("/tmp/mnesia.bak", [])."
+# 6. Enable mining autostart and restore the peers list in the config
+# 7. Start the node
+
+- name: Backup, archive and download Mnesia database
+  hosts: "{{ ansible_play_hosts | last }}"
+  remote_user: epoch
+
+  vars:
+    project_root: "{{ ansible_env.HOME }}/node"
+    download_dir: /tmp/mnesia_backups
+    backup_suffix: backup
+
+  tasks:
+    - name: Check if epoch binary exists
+      stat:
+        path: "{{ project_root }}/bin/epoch"
+      register: epoch
+
+    - name: Ping epoch daemon
+      command: "{{ project_root }}/bin/epoch ping"
+      failed_when: no
+      register: ping
+      when: epoch.stat.exists
+
+    - name: Backup Mnesia database
+      command: "{{ project_root }}/bin/epoch eval 'mnesia:backup(\"/tmp/mnesia_{{ ansible_hostname }}_{{ backup_suffix }}\").'"
+      register: backup
+      when: 
+        - epoch.stat.exists
+        - ping.stdout == "pong"
+
+    - name: Archive Mnesia backup
+      archive:
+        path: /tmp/mnesia_{{ ansible_hostname }}_{{ backup_suffix }}
+        dest: /tmp/mnesia_{{ ansible_hostname }}_{{ backup_suffix }}.gz
+      when: 
+        - epoch.stat.exists
+        - ping.stdout == "pong"
+        - backup.stdout == "ok"
```

Comment:
Does it mean that if `mnesia:backup` returned `{error, Reason}` then playbook succeeds (i.e. only some "skipped" so passed and exit code 0)?